README.md

```diff
@@ -145,7 +170,7 @@ code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
+python TSM.py
```

Comment:
# 5

You produced a parser, but it isn't based on a state machine, and in fact illustrates exactly why a state machine is a good idea. There are flags (such as `noun`) that carry state in an ad hoc way, and there are assumptions (such as the maximum number of adjectives is 2) that aren't true.

A state table approach would look something like:

~~~ py
import unittest
from functools import reduce
ERROR  = "ERROR"
OK     = "OK"

states = {
  0:  { "the":    1 },
  1:  { "lazy":   2, "smelly":  2, "dog": 3, "cat": 3 },
  2:  { "lazy":   2, "smelly":  2, "dog": 3, "cat": 3 },
  3:  { "ate":    4, "ran":     4 },
  4:  { "slowly": 5, "noisily": 5, "EOI": OK },
  5:  { "EOI":    OK }
}

def getNextState(currentState, nextWord):
    return states.get(currentState,{}).get(nextWord, ERROR)

def stateMachine(sentence):
    return reduce(getNextState, sentence, 0) == OK
~~~